src/qpmodel.jl

```diff
@@ -262,20 +266,10 @@ function NLPModels.hess_structure!(
   rows::AbstractVector{<:Integer},
   cols::AbstractVector{<:Integer},
 )
-  if typeof(qp.data.H) <: SparseMatrixCOO
-    rows .= qp.data.H.rows
-    cols .= qp.data.H.cols
-  else
-    nvar = qp.meta.nvar
-    idx = 1
-    for j = 1:nvar
-      for i = j:nvar
-        rows[idx] = i
-        cols[idx] = j
-        idx += 1
-      end
-    end
-  end
+  typeof(qp.data.H) <: SparseMatrixCOO || 
+    error("hess_structure! should be used only if H is a SparseMatrixCOO")
+  rows .= qp.data.H.rows
+  cols .= qp.data.H.cols
```

Comment:
Here again I think it would be more powerful to parametrize `QuadraticModel` and use dispatch.